lib/AST/ModuleNameLookup.cpp

```diff
@@ -118,130 +214,147 @@ static ResolutionKind recordImportDecls(LazyResolver *typeResolver,
 
     // Update the overload set.
     bool stillOverloadable = updateOverloadSet(overloads, newDecls);
-    return stillOverloadable ? ResolutionKind::Overloadable
-                             : ResolutionKind::Exact;
+    if (!stillOverloadable)
+      resolutionKind = ResolutionKind::Exact;
+    break;
   }
 
   case ResolutionKind::Exact:
     // Add all decls. If they're ambiguous, they're ambiguous.
```

Comment:
Again, the second sentence seems like it's trying to say more than it does. Could you take a whack at a clarification?